### PR TITLE
New unit tests

### PR DESCRIPTION
**Pull request by BrightStraightReality**
new unit tests location, drone, flight boundary container, rectangle, point

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@
 *.pyc
 
 *.log
+#hi
```
Comment (DarkmatterVale) [inline .gitignore]: Please remove this

File: README.md [hunk 2]
```diff
@@ -60,6 +60,7 @@ To run the Client script, cd into the ```gcs``` directory, then run:
 ```
 export FLASK_APP=flask_gcs.py
 python -m flask run --with-threads
+
```
Comment (DarkmatterVale) [inline README.md]: Remove this new line

File: tests/unit_tests/test_drone.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import unittest
+
+class DroneTestCase(unittest.TestCase):
+
+    def setUp(self):
+        self.drone = Drone(np.array([0,0,0]), np.array([[ 7, 10, 13], [21, 24, 27]]))
+
+    def test_set_drone_pos(self):
+        self.drone.set_drone_position(np.array([1,1,1]))
+        self.assertEquals(np.array([1,1,1]).any(), self.drone.point.any())
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_drone.py]: This is incorrect. You need to do array_equals here

File: tests/unit_tests/test_drone.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+    def test_add_waypoint(self):
+        self.drone.add_waypoint(np.array([20,20,20]))
+        self.assertEquals(np.array([[ 7, 10, 13], [21, 24, 27], [20, 20, 20]]).any(), self.drone.waypoint_holder.waypoints.any())
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_drone.py]: See above

File: tests/unit_tests/test_drone.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+    def setUp(self):
+        self.drone = Drone(np.array([0,0,0]), np.array([[ 7, 10, 13], [21, 24, 27]]))
+
+    def test_set_drone_pos(self):
+        self.drone.set_drone_position(np.array([1,1,1]))
+        self.assertEquals(np.array([1,1,1]).any(), self.drone.point.any())
+
+    def test_add_waypoint(self):
+        self.drone.add_waypoint(np.array([20,20,20]))
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_drone.py]: You should remove all waypoints before you call this. The tests do not always execute from top of the file to the bottom, so you should clear the waypoints to ensure that the test completes properly.

File: tests/unit_tests/test_drone.py (hunk 4 above)
Comment (DarkmatterVale) [reply tests/unit_tests/test_drone.py]: Additionally, you should only be testing for one waypoint here, not to see if the waypoints contain the added waypoint

File: tests/unit_tests/test_drone.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        self.assertEquals(np.array([[ 7, 10, 13], [21, 24, 27], [20, 20, 20]]).any(), self.drone.waypoint_holder.waypoints.any())
+
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_drone.py]: You should add a second test which adds a second waypoint to the waypoint holder. You should then verify that the first waypoint is still located in the 0th index, and the new waypoint is in the 1st index

File: tests/unit_tests/test_drone.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    def test_reset_waypoints(self):
+        self.drone.reset_waypoints()
+        self.assertEquals(0, self.drone.waypoint_holder.waypoints.shape[0])
+
+    def test_has_reached_waypoint(self):
+        self.drone = Drone(np.array([0,0,0]), np.array([[ 7, 10, 13], [21, 24, 27]]))
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_drone.py]: I would recommend not reinitializing the drone object here. Since you did not do it in any of the other tests, it does not make sense to do it here.

File: tests/unit_tests/test_drone.py [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+        self.assertFalse(self.drone.has_reached_waypoint())
+
+    def test_get_point(self):
+        self.assertEquals(np.array([0,0,0]).all(), self.drone.get_point().all())
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_drone.py]: Again, you need to use the ```array_equals``` method here to compare the arrays. Additionally, you should set the drone point to ```[0, 0, 0]``` at the start of this test to make sure that the drone is actually positioned at that location

File: tests/unit_tests/test_drone.py [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+    def test_get_waypoint_holder(self):
+        self.assertEquals(np.array([[ 7, 10, 13], [21, 24, 27]]).any(), self.drone.get_waypoint_holder().waypoints.any())
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_drone.py]: Again, incorrect numpy compare method

File: tests/unit_tests/test_flight_boundaries_container.py [hunk 10]
```diff
@@ -0,0 +1,12 @@
+from SDA import *
+import numpy as np
+from SUASSystem import *
+import unittest
+
+class FlightBoundaryContainerTestCase(unittest.TestCase):
+
+    def setUp(self):
+        self.boundary = FlightBoundariesContainer(np.array([np.array([[-2000, -2000], [-2000, 2000], [2000, 2000], [2000, -2000]])]))
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_flight_boundaries_container.py]: The variable name should be more descriptive. Since there is a boundary object, you should instead call this ```self.test_boundary_container```

File: tests/unit_tests/test_flight_boundaries_container.py [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+
+    def test_is_point_in_bounds(self):
+        self.assertTrue(self.boundary.is_point_in_bounds(np.array([0,0,0])))
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_flight_boundaries_container.py]: You need additional tests here. This is a very important method in the SDA process, and we use a complex path at competition. While it is great to test a simple path like ```[[-2000, -2000], [-2000, 2000], [2000, 2000], [2000, -2000]]```, you need to test more complex paths as well (non regular polygons).

Additionally, you need multiple tests for each flight boundary. I'd recommend picking 3-4 points in each boundary path and testing those.

You also need to test multi-boundary flight containers (same class, different parameter length). At competition, we have multiple flight boundaries which are all put into this one class. This class needs to be able to determine if a point exists in any of the boundaries.

File: tests/unit_tests/test_point.py [hunk 12]
```diff
@@ -0,0 +1,46 @@
+import numpy as np
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_point.py]: Please remove the ```as np```. This has been a bad habit in the past, and should not be continued.

File: tests/unit_tests/test_point.py [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+from SyntheticDataset import *
+import unittest
+import math
+
+class PointTestCase(unittest.TestCase):
+
+    def setUp(self):
+        self.xInit = 2
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_point.py]: Please fix the variable casing

File: tests/unit_tests/test_point.py [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+
+    def test_toInt(self):
+        self.point3 = self.point.toInt()
+        self.assertEquals((2,3), (self.point3.toInt().getX(), self.point3.toInt().getY()))
+
+    def test_rotate(self):
+        pivot = Point(0,0)
+        spin = math.pi
+        self.point.rotate(pivot, spin)
+        self.assertEquals((-2.000000,-3.000000), (round(self.point.getX(), 6), round(self.point.getY(), 6)))
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_point.py]: Please add a comment here noting the difference in the actual value and the tested value, and that it was insignificant enough for the test to pass

File: tests/unit_tests/test_rectangle.py [hunk 15]
```diff
@@ -0,0 +1,57 @@
+import numpy as np
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_rectangle.py]: See the above comment for a similar change

File: tests/unit_tests/test_rectangle.py [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+        self.height = 8
+        self.x_initial = 0
+        self.y_initial = 0
+        self.width = 6
+        self.rectangle = Rectangle(self.x_initial, self.y_initial, self.width, self.height)
+
+    def test__repr__(self):
+        self.assertEquals("X: 0 Y: 0 Width: 6 Height: 8", self.rectangle.__repr__())
+
+    def test_get_Height(self):
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_rectangle.py]: Fix method naming to be lowercase please

File: tests/unit_tests/test_point.py [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+        self.assertEquals(5, self.point.getY())
+
+    def test___getitem__(self):
+        self.assertEquals(3, self.point.__getitem__(1))
+
+    def test_getTranslatePoint(self):
+        self.point1 = self.point.getTranslatePoint(Point(3,4))
+        self.assertEquals((1,1), (self.point1.getX(), self.point1.getY()))
+
+    def test_getTranslatedPoint(self):
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_point.py]: Please use proper method naming

File: tests/unit_tests/test_rectangle.py [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+    def test_set_Width(self):
+        self.rectangle.setWidth(100)
+        self.assertEquals(100, self.rectangle.getWidth())
+
+    def test_set_Height(self):
+        self.rectangle.setHeight(100)
+        self.assertEquals(100, self.rectangle.getHeight())
+
+    def test_contains(self):
+        self.assertTrue(self.rectangle.contains(Point(1,1)))
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_rectangle.py]: This test should also contain a more comprehensive set of tests. Please add additional test cases here for other points (outside the rectangle, right outside, right inside)

File: tests/unit_tests/test_rectangle.py [hunk 19]
```diff
@@ -0,0 +1,57 @@
...
+
+    def test_get_Mid_point(self):
+        self.assertEquals((3,4), (self.rectangle.getX() + (self.rectangle.getWidth()/2), self.rectangle.getY() + (self.rectangle.getHeight()/2)))
+
+    def test_intersects(self):
+        self.assertTrue(Rectangle(1, 1, 2, 2))
```
Comment (DarkmatterVale) [inline tests/unit_tests/test_rectangle.py]: I don't think this is actually testing intersection right now...

**Review by DarkmatterVale — CHANGES_REQUESTED** [review]
Overall code is sound. Lots of minor fixes that need to be completed before this can be accepted. Please try to have this done within the next day or two.

File: tests/unit_tests/test_rectangle.py (hunk 16 above)
Comment (BrightStraightReality) [reply tests/unit_tests/test_rectangle.py]: done

File: tests/unit_tests/test_rectangle.py (hunk 15 above)
Comment (BrightStraightReality) [reply tests/unit_tests/test_rectangle.py]: done

File: tests/unit_tests/test_point.py (hunk 17 above)
Comment (BrightStraightReality) [reply tests/unit_tests/test_point.py]: done

File: tests/unit_tests/test_point.py (hunk 12 above)
Comment (BrightStraightReality) [reply tests/unit_tests/test_point.py]: done

File: .gitignore (hunk 1 above)
Comment (BrightStraightReality) [reply .gitignore]: don't know how this got there but done

File: tests/unit_tests/test_drone.py (hunk 3 above)
Comment (BrightStraightReality) [reply tests/unit_tests/test_drone.py]: done

File: tests/unit_tests/test_drone.py (hunk 5 above)
Comment (BrightStraightReality) [reply tests/unit_tests/test_drone.py]: done

File: tests/unit_tests/test_drone.py (hunk 4 above)
Comment (BrightStraightReality) [reply tests/unit_tests/test_drone.py]: done

File: tests/unit_tests/test_drone.py (hunk 6 above)
Comment (BrightStraightReality) [reply tests/unit_tests/test_drone.py]: done

File: tests/unit_tests/test_drone.py (hunk 9 above)
Comment (BrightStraightReality) [reply tests/unit_tests/test_drone.py]: done

File: tests/unit_tests/test_drone.py (hunk 8 above)
Comment (BrightStraightReality) [reply tests/unit_tests/test_drone.py]: done

File: tests/unit_tests/test_drone.py (hunk 7 above)
Comment (BrightStraightReality) [reply tests/unit_tests/test_drone.py]: done

File: README.md (hunk 2 above)
Comment (BrightStraightReality) [reply README.md]: done

File: tests/unit_tests/test_flight_boundaries_container.py (hunk 10 above)
Comment (BrightStraightReality) [reply tests/unit_tests/test_flight_boundaries_container.py]: done

File: tests/unit_tests/test_point.py (hunk 13 above)
Comment (BrightStraightReality) [reply tests/unit_tests/test_point.py]: done

File: tests/unit_tests/test_flight_boundaries_container.py (hunk 11 above)
Comment (BrightStraightReality) [reply tests/unit_tests/test_flight_boundaries_container.py]: done

File: tests/unit_tests/test_point.py (hunk 14 above)
Comment (BrightStraightReality) [reply tests/unit_tests/test_point.py]: done

File: tests/unit_tests/test_rectangle.py (hunk 18 above)
Comment (BrightStraightReality) [reply tests/unit_tests/test_rectangle.py]: done

File: tests/unit_tests/test_rectangle.py (hunk 19 above)
Comment (BrightStraightReality) [reply tests/unit_tests/test_rectangle.py]: done

**Review by DarkmatterVale — APPROVED** [review]
